validator/src/main.rs

```diff
@@ -843,6 +844,17 @@ pub fn main() {
                     "maximum total uncompressed file size of downloaded genesis archive",
                 ),
         )
+        .arg(
+            Arg::with_name("wal_recovery_mode")
+                .long("wal-recovery-mode")
+                .value_name("MODE")
+                .takes_value(true)
+                .help(
+                    "Mode to recovery the ledger db write ahead log. \
+                    Options: tolerate_corrupted_tail_records, absolute_consistency,\
```

Comment:
Try using `.possible_values()` to allow clap to document and validate the user's input for ya:
https://github.com/solana-labs/solana/blob/4b93a7c1f6859d3771d5e6a5ec1a03f763eb12b1/dos/src/main.rs#L110-L118

Reply:
oh cool